runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -83,13 +75,14 @@ public void mergeInto(MultiObjectSMREntry other) {
     @Override
     void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
         super.deserializeBuffer(b, rt);
-
-        int numUpdates = b.readInt();
-        entryMap = new HashMap<>();
-        for (int i = 0; i < numUpdates; i++) {
-            entryMap.put(
-                    new UUID(b.readLong(), b.readLong()),
-                    ((MultiSMREntry) Serializers.CORFU.deserialize(b, rt)));
+        int numStreams = b.readInt();
+        for (int i = 0; i < numStreams; i++) {
+            UUID streamId = new UUID(b.readLong(), b.readLong());
+
+            int updateLength = b.readInt();
+            byte[] streamUpdates = new byte[updateLength];
```

Comment:
This adds an extra copy, but also reduces the memory overhead when not all stream updates are needed. 
If we know all open streams, then we can omit the copy and directly deserialize only the opened streams.

Reply:
Seems like this should be a comment in the code and not just in GitHub?

Reply:
This was a reminder, will add the comments to the source.